docs: Add wp-i18n-twig to Internationalization docs

nlemoine opened the pull request:
Add our WP-CLI package to the docs :)

## Issue
<!-- Description of the problem that this code change is solving -->


## Solution
<!-- Description of the solution that this code changes are introducing to the application. -->


## Impact
<!-- What impact will this have on the current codebase, performance, backwards compatibility? -->


## Usage Changes
<!-- Are there are any usage changes that we need to know about? If so, list them here so that we can integrate it in the release notes and developers know what usage changes are associated to your PR.
-->

## Considerations
<!-- As we do not live in an ideal world it's worth to share your thought on how we could make the solution even better. -->


## Testing
<!-- Are unit tests included? If they need to be written, please provide pseudo code for a scenario that fails without your code, but succeeds with it -->

nlemoine committed:
docs: Add wp-i18n-twig to Internationalization docs

Co-authored-by: Erik van der Bas <erik@basedonline.nl>

Levdbas commented on docs/v2/guides/internationalization.md:
```diff
@@ -41,26 +45,41 @@ You can use sprintf-type placeholders, using the `format` filter:
-### Generating l10n files with Poedit 2
+The Timber team provides a [dedicated WP-CLI package](https://github.com/timber/wp-i18n-twig) that takes care of extracting translations from Twig files.
+
+This main benefit over other tools is that it perfectly integrates with the existing `wp i18n` command, making `.pot` file generation simple and easy.
```
```suggestion
The main benefit over other tools is that it perfectly integrates with the existing `wp i18n make-pot` command, making `.pot` file generation simple and easy.
```

Levdbas commented on docs/v2/guides/internationalization.md:
```diff
@@ -41,26 +45,41 @@ You can use sprintf-type placeholders, using the `format` filter:
+
+This main benefit over other tools is that it perfectly integrates with the existing `wp i18n` command, making `.pot` file generation simple and easy.
+
+Note that the package isn't Timber specific, it will work with any Twig implentation making use of WordPress translation functions.
```
```suggestion
Note that the package isn't Timber specific, it will work with any Twig implementation that also provides WordPress translation functions.
```

Levdbas commented on docs/v2/guides/internationalization.md:
```diff
@@ -41,26 +45,41 @@ You can use sprintf-type placeholders, using the `format` filter:
+
+### Alternatives
+
+You can aslo generate `.pot`, `.po` and `.mo` files, with [Poedit](https://poedit.net/).
```
```suggestion
You can also generate `.pot`, `.po` and `.mo` files, with [Poedit](https://poedit.net/).
```

Levdbas reviewed (commented):
Some minor fixes @nlemoine !

nlemoine commented:
Thanks for your fixes @Levdbas! I guess I wrote this too late 😅 (and you're a far better english writer than I ;))

nlemoine commented:
Just wondering, maybe we could remove those old extraction methods mentioned in the docs (python parser, Twig Gettext Extractor)? Those are a bit outdated and brings more confusion. It may be simpler to directly point out the current best option. We all use `wp i18n make-pot` for a while and it works pretty good so far (far better than my previous gulp based workflow!).

gchtr commented:
> Just wondering, maybe we could remove those old extraction methods mentioned in the docs (python parser, Twig Gettext Extractor)?

Oh yes, I’m all for this. I could never manage to get the python parser or the gettext extractor working. Poedit Pro was always the only thing that worked for me.

I would keep the solution with Poedit 2 in there, because it’s an approach that has worked really well over the years. But I would kick out the other solutions.

nlemoine commented:
@Levdbas What do you think?

Levdbas commented:
All in for some housekeeping here!

nlemoine commented:
@timber/rangers I removed all the outdated parts and updated the docs for v1 too since nothing really changed.

Let me know if this needs more changes or feel free to edit if needed :)
